### ISSUE TITLE
[ALL legacy] android-10_legacy setup

### ISSUE THREAD

**Issue by chris42** [sonyxperiadev/bug_tracker#536]
Hi there,
after Loire has been moved to legacy, I am wondering what that actually means?

Is it just now frozen and no further work will happen? Will it only be supported for the official Sony combinations (last one 4.9/Android 9; last working one was Android 8), or will it be supported on base of the binaries (kernel 4.9 no matter if it is combined with e.g. Android 10)?

I fully understand, that an old platform is not in the focus when hands are few. Hence in no means this question is intended to put pressure on anyone for fixes, just would like to know how it will go on.

**Comment by MarijnS95** [comment]
> or will it be supported on base of the binaries (kernel 4.9 no matter if it is combined with e.g. Android 10)?

That has [already been answered](https://github.com/sonyxperiadev/bug_tracker/issues/494#issuecomment-577577954):
> loire can use the legacy manifest to build Android 10 with kernel 4.9 and reuse the Android 9 vendor release

https://github.com/sonyxperiadev/kernel-sony-msm-4.14-common/issues/8#issuecomment-577711767

As you apparently understood we are indeed massively understaffed when it comes to maintaining this many platforms and devices. We didn't have time to get Loire and Tone rock-solid on 4.9 - it's going to take even more time to do a full bring-up and make it work proper on 4.14, but we need compatible binaries in the first place, which for now is the main reason to keep the platforms on 4.9.

Furthermore we are working on mainlining these devices (running on the latest Linux kernel) once and for all, but this takes a lot of time and care. The more time we spend messing around with drivers and whatnot on 4.14, the longer that is going to take.

**Comment by chris42** [comment]
Ok, understood. Didn't see the issue in the kernel repository.

I can do the following: I had a kernel 4.9 / Android 10 version running on my test device. I adapted @stefanhh0 script for it. Will see that I test it with the latest branch and previous ones. I already figured some problems of Android 9 do still persist in Android 10. Will update the issues accordingly with latest logs and information.
Additionally I can report build problems I encounter (where should I report them?).

Would that help?

**Comment by vknecht** [comment]
I'd be happy to help with 4.9/Pie and mainline Loire improvement
Btw just published my [latest Pie builds for Loire](https://forum.xda-developers.com/showpost.php?p=81695499&postcount=62) devices...
One thing I have in mind is updating brcmfmac and it seemed to improve a bit
(got crash reports related to rpmh instead of wifi), but [did it with git am](https://github.com/vknecht/android_kernel_sony_msm/commits/brcmfmac)...
Any guidance how to do it properly is welcome.

**Comment by MarijnS95** [comment]
@chris42 You should not need that script for Loire/Tone anymore as the goal is to build k4.9 + Android 10 out of the box. @ix5 is [working on this](https://github.com/sonyxperiadev/device-sony-common/pull/666), he might be able to give you an update on the status.
As for reporting issues, I don't think we're ready for that yet. @ix5 is likely fully aware of what still needs to be done and patched up; only after it is merged will such feedback be really helpful.

> Any guidance how to do it properly is welcome.

@vknecht I recommend rebasing your work on top of the commit mentioned in the XDA thread, push that to your fork and link the topmost commit instead. Other than that I haven't looked at the picked commits themselves, @oshmoun and @kholk are the brcmfmac guys.
If this really improves things you might want to PR the pushed branch to sonyxperiadev as draft so that it can be reviewed easily.

`git am` is usually the way to go, though you may find it easier to fetch from a remote and use `git cherry-pick <commit_id> ...` instead.

**Comment by ix5** [comment]
"Legacy" means no official updates and mostly frozen state. We will make an effort to get loire+tone buildable "officially" on Android 10 and k4.9 and then freeze that state. Software Binaries updates are very very unlikely.
That means loire+tone are now in your hands. It's time to step up if you want fixes. Bug reports for legacy are nice, but bug reports without PRs to fix them only take away time better spent elsewhere, tbh.

I've created the needed patches here:
https://github.com/sonyxperiadev/local_manifests/pull/86
https://github.com/sonyxperiadev/device-sony-common/pull/666
https://git.ix5.org/felix/device-sony-sepolicy/commit/ca356bc5abfa6a45ff5b82e8d5c8dc43eff0865d
https://git.ix5.org/felix/device-sony-common/commit/f12d93b221a1ce312e8db1589fd925aa04d51244
https://git.ix5.org/felix/device-sony-common/commit/6a67a1a7f89d87dba1419c70c2abd08c1a35816f
https://git.ix5.org/felix/device-sony-common/commit/f068df37449d04b9286800baf5cec873e991b455
And revert https://github.com/sonyxperiadev/device-sony-common/commit/8b79a2321abe42c9d13540651cbf8a276ec7a2f1
I have just split out a legacy `liblights` with the tri-state-LED commit reverted into my own repo https://git.ix5.org/felix/sony-liblights, see the patch to device-sony-common:
```
ifeq ($(SOMC_KERNEL_VERSION),4.14)
 PRODUCT_PACKAGES += \
     android.hardware.light@2.0-service.sony
else
PRODUCT_PACKAGES += \
    android.hardware.light@2.0-service.sony.legacy
endif
```

**Comment by MarijnS95** [comment]
> "Legacy" means no official updates and mostly frozen state

Keyword being "mostly". All the shared repositories and obviously the underlying AOSP project will keep receiving updates.

> That means loire+tone are now in your hands. It's time to step up if you want fixes. Bug reports for legacy are nice, but bug reports without PRs to fix them only take away time better spent elsewhere, tbh.

Indeed, it is time to draw a line. At this point I'd consider Loire+Tone support from the active contributors around here a favor at best. If you rely on this opensource project for one of these devices and must absolutely run the latest version it is up to you to address the issues, or wait for @bartcubbins who seems to have some 4.14 goodies in the pipeline. Or stick around for the ride to Mainline when the time comes around to it.

**Comment by chris42** [comment]
So I guess it makes sense to gather everything round Loire/Kugo here, as discussions spread are spread a bit. Will be closing my other issues. (@vknecht , @stefanhh0)

Current issues that exist in Pie/Kernel 4.9 and have to be tested in a Android 10/Kernel 4.9 version
- Random reboot during usage (former #433)
- Wifi drop out when device goes idle (former #419)
- Audio problems (#430)

**Comment by aaromakinen** [comment]
Hi, thank you for your awesome work! What is the way to build android 10 to suzu with kernel 4.9 right now since using just the legacy-10 won't work and I guess that the script from stefanhh0 doesn't work either.
I'd mostly need only adb to work.

**Comment by chris42** [comment]
So I got a booting Kugo now on the android-10_legacy/r26 branch. Haven't tested yet, but below is what I did
1. Add PR 262 to switch kernel 4.9 in platform.mk
2. Add PR 666 for device-sony-common
3. Add PR 2188 for WifiHAL
4. Revert local_manifest commit f2bc4d5e1bfd7d4b48d373350b70dac49c70d2af
5. Revert of media HAL change 64cfaf3715ec64e040dc978de7fcba4f07807b04
6. Revert bump graphics.composer to version 2.3 0634f99808583bac9586649da024c1e24d14964b
7. Revert liblights change 8b79a2321abe42c9d13540651cbf8a276ec7a2f1 and patch common-treble.mk
8. Added @ix5 patches from comment above

Will report back after a few tests.

@aaromakinen: I would try @stefanhh0 script, it is normally a good base to start and helps to repeat the builds. As you can see in the list above, there is not a straight forward way on getting the device working, you probably need to amend the script to include the right patches and reverts. I guess for either device you will need these (maybe #3 you could ignore) as otherwise I had problems with the compile or getting into bootloops.

**Comment by ix5** [comment]
Alin set up a proper q-mr1-legacy branch with my changes, see https://github.com/sonyxperiadev/bug_tracker/issues/551#issuecomment-593411995

@vknecht you might wanna email me with regards to brcmfmac and legacy stuff, address is at https://ix5.org/contact/

**Comment by chris42** [comment]
@ix5 and @jerpelea All of the above mentioned reverts and changes seem to be included and more. Thanks a lot for this! Will start a build and see how that works.
Will rename the issue to use it for all legacy devices.

@vknecht FYI: I have your [PR 2188](https://github.com/sonyxperiadev/kernel/pull/2188) running on a phone since the morning. So far it only dropped out of wifi once. So it seems to be more stable.

**Comment by SoftReaper** [comment]
as i've read, here is the new "All goes" for legacy devices, so

I've wiped all my source code and synced back from a clean work space to avoid any complication, but, I don't really undestand what's going on:

```[ 96% 104862/108519] //system/extras/verity:VeritySigner javac [linux_glibc common]
FAILED: out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/VeritySigner.jar
rm -rf "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/classes" "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/anno" "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars" && mkdir -p "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/classes" "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/anno" "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars" && out/soong/host/linux-x86/bin/zipsync -d out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars -l out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars/list -f "*.java"  && (if [ -s out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars/list ] || [ -s out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/VeritySigner.jar.rsp ] ; then out/soong/host/linux-x86/bin/soong_javac_wrapper prebuilts/jdk/jdk9/linux-x86/bin/javac -J-Xmx2048M -Xmaxerrs 9999999 -encoding UTF-8 -sourcepath "" -g -XDskipDuplicateBridges=true -XDstringConcat=inline  -proc:none  -bootclasspath prebuilts/jdk/jdk8/linux-x86/jre/lib/jce.jar:prebuilts/jdk/jdk8/linux-x86/jre/lib/rt.jar -classpath out/soong/.intermediates/external/bouncycastle/bouncycastle-unbundled/linux_glibc_common/javac/bouncycastle-unbundled.jar -source 1.8 -target 1.8 -d out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/classes -s out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/anno @out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/VeritySigner.jar.rsp @out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars/list ; fi ) && out/soong/host/linux-x86/bin/soong_zip -jar -o out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/VeritySigner.jar -C out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/classes -D out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/classes && rm -rf "out/soong/.intermediates/system/extras/verity/VeritySigner/linux_glibc_common/javac/srcjars"
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007ff0a47c586d, pid=225747, tid=225894
#
# JRE version: OpenJDK Runtime Environment (9.0+181) (build 9+181-android4685155)
# Java VM: OpenJDK 64-Bit Server VM (9+181-android4685155, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x96b86d]  PhaseIdealLoop::build_loop_early(VectorSet&, Node_List&, Node_Stack&)+0x8d
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P" (or dumping to /root/android/core.225747)
#
# An error report file with more information is saved as:
# /root/android/hs_err_pid225747.log
#
# Compiler replay data is saved as:
# /root/android/replay_pid225747.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#

#### failed to build some targets (02:17:42 (hh:mm:ss)) ####```

**Comment by MartinX3** [comment]
@SoftReaper
Please try it again with either Java 8 or Java 11.
Java 9 is deprecated / supported anymore.

**Comment by SoftReaper** [comment]
@MartinX3 : it's precisely what i'm not understanding, output from `apt list --installed`:

```
...
ntfs-3g/bionic-updates,bionic-security,now 1:2017.3.23-2ubuntu0.18.04.2 amd64  [installé, automatique]
openjdk-8-jdk/bionic-updates,bionic-security,now 8u242-b08-0ubuntu3~18.04 amd64  [installé]
openjdk-8-jdk-headless/bionic-updates,bionic-security,now 8u242-b08-0ubuntu3~18.04 amd64  [installé, automatique]
openjdk-8-jre/bionic-updates,bionic-security,now 8u242-b08-0ubuntu3~18.04 amd64  [installé, automatique]
openjdk-8-jre-headless/bionic-updates,bionic-security,now 8u242-b08-0ubuntu3~18.04 amd64  [installé, automatique]
openprinting-ppds/bionic,bionic,now 20180306-1 all  [installé, automatique]
...
```

probably the prebuilt version, as it includes JDK9 and 8 packed

i'll try to remove 8 and moving to 11

**Comment by stefanhh0** [comment]
Java-9 comes with the android reps: 9+181-android4685155. Changing the systems java won't have an effect. 

- Is it a reproducible problem or a one time occurrence? 
- Did you try a clean boot of your system before re-doing the build
- Have you done a parallel threaded build? If so, you could try doing a single-threaded build and see if it then still crashes.
- With how many threads are you building and how much memory do you have?
- Other software is running stable on your system, i.e. no other sigsegv occurrences so far?
- Check the logs for your system for any suspicious messages e.g. oom killer or similar.

Although java is pretty stable,  I have seen such crashes from time to time, however then it was gone on a second attempt, so more or less race conditions resp. not easy reproducible vm state constellations that trigger a crash.

**Comment by SoftReaper** [comment]
> Is it a reproducible problem or a one time occurrence?
first time i got it in 3 days full of builds trials

> Did you try a clean boot of your system before re-doing the build
i've booted, done a repo sync, source & lunch then make, , so could be considered as a clean boot

> Have you done a parallel threaded build? If so, you could try doing a single-threaded build and see if it then still crashes.
> With how many threads are you building and how much memory do you have?
18 Threads (make -j18), 11GB of Ram 50GB swap, source on RAID 10 HDD swap on NVME Raid 0

> Other software is running stable on your system, i.e. no other sigsegv occurrences so far?
not at my knowledge, got htop running while building to monitor background processes and memory/swap usage

> Check the logs for your system for any suspicious messages e.g. oom killer or similar.
not in the logs

As it is the first time I've seen this error (for this particular one), I've tried to re update repo, cleaning out folder and restart build with the same conditions and parameters, always seen this step running without even a warning in pevious tries, so just got to wait and see if it happends again

**Comment by stefanhh0** [comment]
make -j18 is too high esp. when only having 11G RAM. Is your CPU anyway capable running 18 threads in parallel? I'd recommend trying it with -j1 to see if that fixes your problem. Then you may increase the number of threads to find the point where it is starting to become unstable. It normally makes not too much sense to increase the number of build threads beyond the cpu-limits.

**Comment by SoftReaper** [comment]
Threadripper 1920x 12c/24t, only 18 assigned to my VM, android Pie did without a trouble with j18, but if the build fail (it'll append in less than 20 minutes if it had to, 97%) i'll try -j1

**Comment by SoftReaper** [comment]
it worked, currently testing:

things looks good for now, a bit "hot", but functionnal

**Comment by chris42** [comment]
So for Loire/Kugo i got the following out of my tests.
Working
- Wifi (regarding #419 it is way more stable with PR2188, only had 1 or 2 dropouts)
- NFC (tag reading works)
- Bluetooth (shows devices, haven't paired it yet)
- Phone calls in and out
- SMS

Not Working
- GPS/location seem to have an issue. Using my other device and satstat, it gets the satellites and position quite quickly (<30sec), the new build however found one satellite, but it took ages (>10min) and no position
- Wifi Hotspot 2.4/5Ghz

To be tested:
- Random reboot during usage (former #433)
- Audio problems (#430)
As both of them only occur after some time, I'll probably need some time as well to figure if there is still a problem

**Comment by stefanhh0** [comment]
> Not Working
> 
> - GPS/location seem to have an issue. Using my other device and satstat, it gets the satellites and position quite quickly (<30sec), the new build however found one satellite, but it took ages (>10min) and no position

I had such issues as well, GPS acquisition can be really slow. You'd wait longer and make the test when it is not too cloudy and you are not shielded by buildings around you or being in a slot-canyon and such.

If you are experiencing only <30sec with other devices than that is because of the aGPS functionality. First initialization takes usually long. That is what @jerpelea explained to me some time ago. If the problem persists, you'd open a separate issue with logs and so on. I can't judge on the hardware in Loire and Yoshino but wouldn't be surprised if they are not among the best what is available on the market when it comes to GPS signal aquisition speed :-)

**Comment by chris42** [comment]
> If you are experiencing only <30sec with other devices than that is because of the aGPS functionality. 

That is what I wonder. Does the GPS need the agps data to receive satellite data?
The other device gets satellite signals (GPS, Glonass, Beidou seems to be too weak) nearly instantly after opening satstat.

Will let it search for a while and report back. Reflashing today with the newer kernel probably didn't help to have a long initialization. ;-)

**Comment by SoftReaper** [comment]
i'm not thinking this way, i've used the "GPS Test" app from google play:
https://play.google.com/store/apps/details?id=com.chartcross.gpstest&hl=fr

i've used it to "force" A-GPS update, but location didn't worked in maps (web browser and app), and it didn't show using wifi location data when using wlan, so i'm thinking it's more a problem of location service.

Moreover, i saw some information about ignored commit under ./repo_update.sh including qcom/gps, if qualcomm provided more than drivers/binaries on the table, it could be a clue

**Comment by mirh** [comment]
GPS is also wanky on kagura. 
And whether they are the latest and shiniest hardware, shouldn't matter to get at least <10 seconds locks.

**Comment by chris42** [comment]
FYI an update to Kugo/Loire: After 4 days, I finally got a GPS lock (and now getting a lock again takes ~15-20min). So it seems to work, just very slow. I compared it with my 8.1, in both I erased the a-GPS data via GPS Test.The 8.1 took about 1-2 hours before the first location and now somewhat below 10sec.

I finally paired and used a bluetooth headset and it is working as well. 
Also didn't have any reboots or audio problems so far, so it seems to be more stable, but I will probably only get a final confirmation when I switch it to use in my daily driver.
The wifi still loses the connection from time to time, but I am no driver expert to tell what is going wrong. Posted some logs to [PR 2188](https://github.com/sonyxperiadev/kernel/pull/2188)

**Comment by chris42** [comment]
I just tried to compile _r30 and it seems, that some more rules by google on how to compile are effective now. You get
`kernel/sony/msm-4.9/kernel/Android.mk:267: error: PHONY target "out/target/product/kugo/kernel" looks like a real file (contains a "/")`

As I understood, this is a problem with the old kernel build setup. As far as I got, the kernel Android.mk ([PR2119](https://github.com/sonyxperiadev/kernel/pull/2119)) and drivers/input/touchscreen/siw/mon/Android.mk (lost the PR, I think also form @kholk ) have to go. Then actually the build will continue, however stop then as it can't find the kernel itself.
`ninja: 'out/target/product/kugo/kernel', needed by 'out/target/product/kugo/boot.img', missing and no known rule to make it`

It works, if you copy the kernel over manually, however somewhere the copy must have gone. I found a few changes by @ix5 and moving targets from devices to common, but not sure, if it is the right place to look.
When continuing compiling, I then end up on this
```
FAILED: 
external/wpa_supplicant_8/hostapd/Android.mk: error: "hostapd (EXECUTABLES android-arm) missing lib_driver_cmd_simulated (STATIC_LIBRARIES android-arm)" 
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
external/wpa_supplicant_8/hostapd/Android.mk: error: "hostapd_nohidl (EXECUTABLES android-arm) missing lib_driver_cmd_simulated (STATIC_LIBRARIES android-arm)" 
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
external/wpa_supplicant_8/wpa_supplicant/Android.mk: error: "wpa_supplicant (EXECUTABLES android-arm) missing lib_driver_cmd_simulated (STATIC_LIBRARIES android-arm)" 
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
frameworks/opt/net/wifi/libwifi_hal/Android.mk: error: "libwifi-hal (SHARED_LIBRARIES android-arm) missing libwifi-hal-emu (STATIC_LIBRARIES android-arm)" 
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this is intentional, but that may defer real problems until later in the build.
build/make/core/main.mk:1008: error: exiting from previous errors.
22:42:26 ckati failed with: exit status 1
build/make/core/main.mk:21: recipe for target 'run_soong_ui' failed
```
Here I am pretty lost. Anyone got a clue?

**Comment by MarijnS95** [comment]
@chris42 As for the kernel image, make sure you have the prebuilt repository checked out at the expected location. There may be other variables involved, please trace the codepath down in common.

As for the build errors, those are pretty common because some of them are set by our device repos. They simply mean you forgot to `lunch` and are building some "broken" generic device.

**Comment by chris42** [comment]
@MarijnS95 Thanks for the quick response! I actually haven't changed the checkout location. Will poke around a bit more in common and see what is missing.

As for the build. You are right, lunch is missing. Apparently it gets lost after a stopped build. Will make sure to only use my script that late...

**Comment by stefanhh0** [comment]
I can confirm the problem, something has changed... wanted to build 4.9.216 bootimage, but now I am running in to the same problem that the build system no longer picks up the prebuilt kernel.

**Comment by chris42** [comment]
I found the following commits so far, that seem to move around the output configuration, but something is still missing (I guess).
For device-loire 
[678af16ec3d2283e0a88da9130d623d60cffa6b2](https://github.com/sonyxperiadev/device-sony-loire/commit/678af16ec3d2283e0a88da9130d623d60cffa6b2)
and in "master" of device-common
[b0e673d01808bf5ab8fb09a337dfb6ffa8b96496](https://github.com/sonyxperiadev/device-sony-common/commit/b0e673d01808bf5ab8fb09a337dfb6ffa8b96496)
[e79867aa472c973dba0e44ca626617ab3d5d9b07](https://github.com/sonyxperiadev/device-sony-common/commit/e79867aa472c973dba0e44ca626617ab3d5d9b07)

However the last two commits are missing in the "q-mr1-legacy" branch. I am not sure, if that's it, as I didn't have time yet to test them and also couldn't see the path (out of the error) in the changes.

**Comment by stefanhh0** [comment]
Those two commits look promising, can you cherry-pick them for the legacy branch and check if it works then? If those fix the problem please open a PR for the legacy-branch.

**Comment by chris42** [comment]
Yepp, already running a test right now... but will take some time...

**Comment by chris42** [comment]
Created draft PRs 
[kernel/PR2216](https://github.com/sonyxperiadev/kernel/pull/2216)
[device-sony-common/PR704](https://github.com/sonyxperiadev/device-sony-common/pull/704)

**Comment by chris42** [comment]
So put both PRs in ready for review. Did a compile for Kugo and it works for me.

**Comment by maluus** [comment]
Thanks for notifying Chris. I've been using stock firmware lately while waiting Android 10 and you know 9+4.9 wasn't really good as daily since this XC is my only phone. If there won't be a 4.14 and we are now suck with 4.9 then I can begin compiling again but I m a bit off from all the recent developments. Were you able to get 10+4.9 combination working?

**Comment by chris42** [comment]
@maluus 
Yes, I got it working (based on the script of @stefanhh0 ) and right now I am looking into getting a sort of stable version to use for my daily driver as well. So far my test device seems to work better than the Android 9 version before. So maybe to update the status for Kugo/Loire:
- Issue #419 Wifi disconnect in standby: This seems to be better with [PR2188](https://github.com/sonyxperiadev/kernel/pull/2188) but still I encountered some rare dropouts (I can't investigate further, as this is beyond my capabilities...)
- Issue #433 Random reboots: Did not have a reboot yet, but as they were very infrequent, it is hard to tell
- Issue #430 Sound problem: So far sound on the test device is working without interruption
- Issue #515 phone app crash: I found that in the logs as well (not regularly). Just trying a build to see if the proposed fix works.

Right now I am compiling _r30 with common/PR704, kernel/PR2188, 2210 and 2216 (Hope all of them will get merged).
Happy to have your feedback and findings in here as well!

**Comment by maluus** [comment]
Good to here. I as I mentioned in my previous post. I need to prepare my build environment. I will try to get my build environment as soon as possible but it would be nice if you can share your build with me?

**Comment by chris42** [comment]
As soon as I have a clean one, I can do that (although it is a very limited thing without gapps, etc.)
@vknecht also published a build in [PR2210](https://github.com/sonyxperiadev/kernel/pull/2210)

**Comment by maluus** [comment]
Plase do that. I thought opengapps would do the trick. Or are there problems with it too?

**Comment by chris42** [comment]
Not that I know of, I am just not using it.

**Comment by maluus** [comment]
Any progress chris?

**Comment by chris42** [comment]
I have my latest build running since Monday now. Wifi is way better, but got one crash. The audio was stable so far, but it sometimes took several days to show. Otherwise the device was working normally.

Not sure if my build helps you (as said no gapps, so 99% of apps from play store won't work), however I put my build environment [here](https://github.com/chris42/android_build). 
With the docker and docker-compose, it should be a matter of minutes to be up and running. 
To use the build script, you would need to adapt it to your needs, e.g. integration of opengapps (you can have a look on how @stefanhh0 did it [here](https://github.com/stefanhh0/aosp-10))

**Comment by chris42** [comment]
So bad news (thought last week was Friday the 13th ;-)): I got the crashes (reboot when using it) and the audio problems (sound goes away after some time) as they were on 9.1. I guess that was to expected, however always good to try.

As with 9.1: without help I can't solve that. I got some logs and pstore, however don't know where to look. This is way beyond my capabilities, sorry.

**Comment by maluus** [comment]
So we don't have any other choice than wait for @bartcubbins to get 4.14 working if he still works on it. Btw was the 4.4 stable for pie? I ll try either 4.4 with pie or aosp 8.1. Got original firmware backup. So some trying won't hurt :D

**Comment by chris42** [comment]
I don't think he can get 4.14 running without proper vendor binaries? As @MarijnS95 and @ix5 - understandably - explained, there are just not enough hands to support Kugo/Loire transfer to 4.14.

8.1 with the 4.4 Kernel was running well, never tried pie with it.
It is probably better to channel any effort in Android 10 with 4.9 than trying new combinations. Especially as only Android 10 will give you updates for the next few months.

**Comment by maluus** [comment]
The problem is to do with the 4.9 kernel and we were trying to figure out and solve the problem even when 4.9 was actively used kernel but to no avail. Now platform dropped from active development and so did the 4.9. It doesn't look promising honestly. Personally I won't be meddling with anymore unless something magical happens :D I'll decide either official firmware or AOSP 8.1 and stick with it.

**Comment by chris42** [comment]
I was hoping to get it sort of stable on a 10/4.9 combination. Without a compact device alternative it sucks even more. :-(

**Comment by maluus** [comment]
if you upload your builds somewhere i may give it a try.

**Comment by chris42** [comment]
Try this [one](https://temp-dlfi23hr2iqvc743ihf.s3.eu-central-1.amazonaws.com/aosp_f5321-img-eng.root.zip)
Let me know when you got it, then I'll remove the link. Does not want to become a ROM builder by accident. Of course you use it on your own risk. ;-)

**Comment by maluus** [comment]
got it thanks

**Comment by maluus** [comment]
btw. is this eng build. if I remember correctly eng builds causes some problems. why not userdebug?

**Comment by chris42** [comment]
I know, always did it this way, what problems came up? O_o

**Comment by maluus** [comment]
Don't remember exactly. I just remember that I read somewhere that eng builds cause stability problems.

**Comment by chris42** [comment]
How is the build working for you? Any new issues or just the ones we know about?

**Comment by maluus** [comment]
Btw I would like to discuss some issues about set up and build problems also microg but don't have your e-mail. can you contact me over fardic@gmail.com

**Comment by rogora** [comment]
If I remember with the eng build the settings app crashes when you open a specific menu, I think it was "Developer options".
Anyway, just to add some feedback, I've been using Android10_r26 with kernel 4.9.213 with PR 2188 for some days as my daily driver and:
- I see random reboots, maybe once a day
- the audio never stopped working like it did with Android 9
- wifi seems to be reliable, but I couldn't test network switch properly due to covid
- I cannot use my headphones, it looks like they're completely ignored, but haven't investigated on this

**Comment by MarijnS95** [comment]
> Don't remember exactly. I just remember that I read somewhere that eng builds cause stability problems.



> If I remember with the eng build the settings app crashes when you open a specific menu, I think it was "Developer options".

For reference: https://github.com/sonyxperiadev/bug_tracker/issues/185#issuecomment-427132369

**Comment by chris42** [comment]
Developer menu works without issues, hence I guess that error with the oem_lock was fixed.
Also tested headphones just now, they worked without issues. Maybe some pocket lint?

**Comment by MarijnS95** [comment]
> hence I guess that error with the oem_lock was fixed.

Yeah [that happened](https://android.googlesource.com/platform/packages/apps/Settings/+/7b1ec94fc97fd4c13f84e06c75683b25b4061b4b%5E%21/src/com/android/settings/development/OemUnlockPreferenceController.java).

**Comment by rogora** [comment]
> Also tested headphones just now, they worked without issues. Maybe some pocket lint?

Indeed, apparently it was some kind of dust in the jack! Sorry for the deceptive report :)

**Comment by chris42** [comment]
Just a small improvement, found a solution to prevent the WifiHAL logspam on Kugo
```
03-08 10:43:02.616   389   389 E WifiHAL : wifi_get_link_stats: requestResponse Error:-3
03-08 10:43:02.617  2030  2206 E WifiVendorHal: getWifiLinkLayerStats(l.937) failed {.code = ERROR_NOT_SUPPORTED, .description = }
```
A framework patch, but whoever wants to add it, find it at [LineageOS](https://review.lineageos.org/c/LineageOS/android_frameworks_opt_net_wifi/+/260124)
I tested it on Kugo and it does reduce the logspam

**Comment by chris42** [comment]
So regarding crashes, I think I got logging on what happens. These are the last lines before reboot. The rest of the log shows, it goes on like this for a while before. Had same stuff in pstore before:
```
03-31 01:35:15.200     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.200     0     0 E msm_rpm_send_data: Failed to write data msg_size:40 ret:0 msg_id:5015
03-31 01:35:15.200     0     0 E msm_rpm_wait_for_ack: Invalid msg id
03-31 01:35:15.200     0     0 E rpm_vreg_send_request: pm8950_s6_level_ao: msm rpm send failed: smpa 6; set=act, rc=-12
03-31 01:35:15.200     0     0 E rpm_vreg_set_voltage: pm8950_s6_level_ao: set voltage for key=vlvl failed, rc=-12
03-31 01:35:15.200     0     0 E cpr_mx_set: apc0_corner: set: vdd_mx [corner:6, fuse_corner:3] = 384 uV failed: rc=-12
03-31 01:35:15.200     0     0 E clk     : clk_core_set_rate_nolock: failed to set a53ssmux clock to run at 1305600000
03-31 01:35:15.202     0     0 E flush_clk_data: Failed to set_rate 19000000 for 2048
03-31 01:35:15.202     0     0 E msm_bus_commit_data: Err flushing clk data for:2048
03-31 01:35:15.202     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.203     0     0 I chatty  : uid=0(root) logd identical 3 lines
03-31 01:35:15.203     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.203     0     0 E msm_rpm_send_data: Failed to write data msg_size:40 ret:0 msg_id:5016
03-31 01:35:15.203     0     0 E msm_rpm_wait_for_ack: Invalid msg id
03-31 01:35:15.204     0     0 E clk     : clk_core_set_rate_nolock: failed to set mmssnoc_axi_a_clk clock to run at 19000000
03-31 01:35:15.204     0     0 E clk     : clk_core_set_rate_nolock: failed to set sysmmnoc_msmbus_a_clk clock to run at 19000000
03-31 01:35:15.204     0     0 E setrate_nodeclk: failed to setrate clk
03-31 01:35:15.204     0     0 E flush_clk_data: Failed to set_rate 19000000 for 2048
03-31 01:35:15.204     0     0 E msm_bus_commit_data: Err flushing clk data for:2048
03-31 01:35:15.205     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.205     0     0 I chatty  : uid=0(root) logd identical 3 lines
03-31 01:35:15.205     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.206     0     0 E msm_rpm_send_data: Failed to write data msg_size:40 ret:0 msg_id:5017
03-31 01:35:15.206     0     0 E msm_rpm_wait_for_ack: Invalid msg id
03-31 01:35:15.206     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.206     0     0 I chatty  : uid=0(root) logd identical 3 lines
03-31 01:35:15.206     0     0 E smd_write_start: packet of size: 20 in progress
03-31 01:35:15.206     0     0 E msm_rpm_send_data: Failed to write data msg_size:40 ret:0 msg_id:5018
03-31 01:35:15.206     0     0 E msm_rpm_wait_for_ack: Invalid msg id
03-31 01:35:15.206     0     0 E rpm_vreg_send_request: pm8950_s2_level: msm rpm send failed: smpa 2; set=act, rc=-12
03-31 01:35:15.206     0     0 E rpm_vreg_set_load: pm8950_s2_level: set mode failed, rc=-12
03-31 01:35:15.206     0     0 E pm8950_s2_level: failed to set load 100000
03-31 01:35:15.206     0     0 E subsys-pil-tz 1de0000.qcom,venus: Failed to set regulator mode(rc:-12)
03-31 01:35:15.208     0     0 E rpm_vreg_send_request: pm8950_s6_level_ao: msm rpm send failed: smpa 6; set=act, rc=-12
03-31 01:35:15.208     0     0 E rpm_vreg_set_voltage: pm8950_s6_level_ao: set voltage for key=vlvl failed, rc=-12
```
I figured that @kholk was working on the respective code, so if you have time, it would be great if you could take a look here. I am not sure if it is connected to your [commit regarding waiting for ACK](https://github.com/sonyxperiadev/kernel/commit/f0e1f8fc628c9cd0667e9719603ca37d290bd5fb#diff-ea41c2af674533c0bcfb5c7fa9fd11aa)

**Comment by chris42** [comment]
Oh @kholk If this could be misunderstood: I am not saying you broke something, but you are the one who understands what is happening.

**Comment by kholk** [comment]
The RPM sometimes doesn't get ready in time at boot because of a bootloader issue... This was getting addressed in some very very very hacky way on previous kernels, but that's no more in 4.9.

Kernel 4.14 has a totally different (and stable) RPM messaging framework, which is the reason why I didn't spend a week of time on this particular issue here on 4.9 (of course I was expecting to have more time to get 4.14 ready on loire).

The issue though happens only at boot, so after the phone is able to get a good boot this thing doesn't happen anymore for the whole OS life.

Regarding the support for legacy devices on legacy kernel, well, our issue is time!
I'd really like to have some to fix everything here but please understand that I'm working on all devices and something has more priority than something else (a totally not working platform or a partially working one has priority on one that has "fine-ish" bugs).

I cannot promise anything at this stage, also because at the moment I don't even have my workstation as I'm not in my city.

Hard times!

**Comment by chris42** [comment]
@kholk Thanks for the explanation and helping me understand what is going on here! 
I am getting the errors also during runtime, when the device crashes. Might tinker with it a bit and see if it improves. Right now I have a device running with reverted changes in rpm-smd.c and it seems to be fine for now. The random crashes however need more runtime (weeks) to see if they occour.

Next I wanted to check on other builds that use different kernels. So hope it is worthwhile to compare what they did.

Fully understand the time issues and thanks a lot for all the work you have put in over the time! 
I am not expecting much for loire, just hope I can keep it running till I find a compact replacement.

Stay safe!

**Comment by chris42** [comment]
Loire/Kugo seems to work without with _r33 of newest android updates.

**Comment by chris42** [comment]
Just build Loire/Kugo with _r36. Seems to work fine (Wifi, NFC, Bluetooth, Phone).

**Comment by chris42** [comment]
So just a short update: this pull request seems to solve the random reboots during use on my Kugo: https://github.com/sonyxperiadev/kernel/pull/2262

I've tested this for a week now and device runs stable. Apparently the higher frequency will use 50mV more power that was missing in the current config. Removing the higher frequency solves this.

Will now test another PR that adds 50mV and keeps the higher frequency.

**Comment by chris42** [comment]
The step with +50mV works as well, but will need to check stability over time. However this is overclocking/overvolting and will not be included in the official sony repositories.
If you want to use it at your own risk, you need to revert https://github.com/sonyxperiadev/kernel/pull/2262 and add https://github.com/sonyxperiadev/kernel/pull/2266

**Comment by chris42** [comment]
Probably makes sense to close this one now with the end of Android 10.

To sum up the last issues with it:
- Reboots are very very rare
- Sound still drops out after some time
- Tethering does not work
- Kernel sources will need updating to latest revisions.

**Comment by MarijnS95** [comment]
> Tethering does not work

FWIW tethering is broken on all devices, and we have yet to invest time why. Fix it on Android 10/11 + k4.14 and it should be backportable to 10/k4.9 as well. Otherwise we'll get it on the mainline kernel "at some point".

> Kernel sources will need updating to latest revisions.

@stefanhh0 Has been taking care of submitting Linux stable patches to sonyxperiadev/kernel?

**Comment by chris42** [comment]
> @stefanhh0 Has been taking care of submitting Linux stable patches to sonyxperiadev/kernel?

I did the last ones for 4.9 up to .230. However merge conflicts are getting to complicated for me to safely go on.

**Comment by MarijnS95** [comment]
Ah, I totally misremembered the author, sorry!

Perhaps you can open a PR and ask advice in there. By chance the same backports that caused conflicts on 4.14 are backported to 4.9, causing the same conflicts there too.

**Comment by jaylinski** [comment]
This may be a stupid question, but can/should I use `android-security-10.0.0_r63` to build a version with the latest security patches?

I'm confused by https://source.android.com/setup/start/build-numbers, because it lists `android-security-10.0.0_r63` with patch level `2019-09-05`, but `android-10.0.0_r42` with patch level `2020-05-05`.

Can someone educate about the connection between the revision and the patch level?

**Comment by jerpelea** [comment]
please use android-10.0.0_r42
